src/fparser/two/Fortran2008.py

```diff
@@ -96,6 +101,241 @@ class Program_Unit(Program_Unit_2003):  # R202
     subclass_names.append("Submodule")
 
 
+class Component_Attr_Spec(STRINGBase):  # R437
+    """
+:F08R:`437`
+    <component-attr-spec> = <access-spec>
+                            | ALLOCATABLE
+                            | CODIMENSION <lbracket> <coarray-spec> <rbracket>
+                            | CONTIGUOUS
+                            | DIMENSION ( <component-array-spec> )
+                            | POINTER
+
+    This rule adds CODIMENSION and CONTIGUOUS attributes to Fortran2003's R441.
+    """
+    subclass_names = ['Access_Spec', 'Dimension_Component_Attr_Spec',
+                      'Codimension_Attr_Spec']
+    use_names = []
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for component attribute specifications.
+
+        :param str string: the string to match as an attribute.
+        :return: None if there is no match, otherwise an object with the \
+            matched attribute.
+        :rtype: None, Component_Attr_Spec, Codimension_Attr_Spec, \
+            Access_Spec, Dimension_Component_Attr_Spec
+        '''
+        return STRINGBase.match(
+            ['ALLOCATABLE', 'CONTIGUOUS', 'POINTER'], string)
+
+
+class Codimension_Attr_Spec(WORDClsBase):  # R502.d
+    """
+    <codimension-attr-spec> = CODIMENSION <lbracket> <coarray-spec> <rbracket>
+    """
+    subclass_names = []
+    use_names = ['Coarray_Bracket_Spec']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for the CODIMENSION attribute.
+
+        :param str string: the string to match as the attribute.
+        :return: None if there is no match, otherwise the matched object.
+        :rtype: None, Codimension_Attr_Spec
+        '''
+        return WORDClsBase.match(
+            'CODIMENSION', Coarray_Bracket_Spec, string, colons=False,
+            require_cls=True)
+
+
+class Coarray_Bracket_Spec(BracketBase):  # R502.d.0
+    """
+    <coarray-bracket-spec> = <lbracket> <coarray-spec> <rbracket>
+    """
+    subclass_names = []
+    use_names = ['Coarray_Spec']
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for the coarray specification
+        including the square brackets.
+
+        :param str string: the string to match as the specification.
+        :return: None if there is no match, otherwise the matched object.
+        :rtype: None, Coarray_Bracket_Spec
+        '''
+        return BracketBase.match('[]', Coarray_Spec, string)
+
+
+class Attr_Spec(STRINGBase):  # R502
```

Comment:
I think it would be simpler to implement this as a subclass of Fortran2003.py's R503 (see Program_Unit in Fortran2008.py as an example).

Reply:
Makes sense for consistency but would only change the definition of `subclass_names`. The match is done using a pattern and requires reimplementation for F08.

Reply:
OK.